### PR TITLE
[8.11.x] PLANNER-2474 optaplanner CI (#1440) 

### PR DESCRIPTION
**Pull request by radtriste**
# JIRA
https://issues.redhat.com/browse/PLANNER-2474

# Cherry-pick
https://github.com/kiegroup/optaplanner/pull/1440

# Related PRs:
* https://github.com/kiegroup/kogito-pipelines/pull/246
* https://github.com/kiegroup/optaplanner/pull/1529
* https://github.com/kiegroup/optaplanner-quickstarts/pull/165
* https://github.com/kiegroup/optaweb-employee-rostering/pull/681
* https://github.com/kiegroup/optaweb-vehicle-routing/pull/551

### CODE REVIEW

**Comment by radtriste** [comment]
jenkins retest this

**Comment by radtriste** [comment]
jenkins retest this

File: .github/workflows/full-downstream.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (radtriste) [inline .github/workflows/full-downstream.yml:1]: I guess this file and the pull-request ones need to be adjusted or ?
@Ginxo could you do so ? or any backport from `main` branch to be done ?

File: .github/workflows/full-downstream.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - '.ci/**'
+
+jobs:
+  concurrency:
+    group: pull_request-${{ github.head_ref }}
+    cancel-in-progress: true
```
Comment (rsynek) [inline .github/workflows/full-downstream.yml:21]: I think this part should go unto the `build-chain` job; if I remember correctly, this is the configuration issue we had on the `main` branch.
Also, the group should be different than what we use in the `pull-request.yml`.

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - '.ci/**'
+
+jobs:
+  concurrency:
+    group: pull_request-${{ github.head_ref }}
+    cancel-in-progress: true
```
Comment (rsynek) [inline .github/workflows/pull_request.yml:21]: ditto

**Review by rsynek — CHANGES_REQUESTED** [review]
Looks good, but the .yml files backport the configuration issue as well.

**Comment by radtriste** [comment]
@Ginxo you are the GHAction expert :) can you have a look to Radovan comments ?

File: .github/workflows/full-downstream.yml (hunk 2 above)
Comment (radtriste) [reply .github/workflows/full-downstream.yml]: this is set like that on `main`: https://github.com/kiegroup/optaplanner/blob/main/.github/workflows/full-downstream.yml#L23

Comment (radtriste) [reply .github/workflows/full-downstream.yml]: oh right it should be inside the build-chain. let me correct that

Comment (radtriste) [reply .github/workflows/full-downstream.yml]: cherry-picking solved it

**Comment by radtriste** [comment]
jenkins retest this

**Comment by radtriste** [comment]
jenkins rerun examples tests

**Comment by radtriste** [comment]
jenkins rerun kogito-examples tests

**Comment by radtriste** [comment]
examples tests failing will be corrected in https://github.com/kiegroup/kogito-examples/pull/900

**Pull request merged**